storage.go

```diff
@@ -356,3 +358,21 @@ func (s *storageImpl) IterateDocs(q *Query, consumer docConsumer) error {
 	}
 	return s.iterateDocsSlice(q, consumer)
 }
+
+func (s *storageImpl) ListCollections() ([]string, error) {
```

Comment:
This implementation is not iterating properly on the collection names.
Basically, you are iterating on all the key/value pairs contained in the database.
The test you wrote `TestListCollection()` works only because you start from an empty database, so the only key you find are keys related to collections creation. Try to specify a valid collectionPath in the `runCloverTest()` (for example todosPath) instead of the empty string, to see what happens.

To iterate on only the correct key/value pairs you should write something like this:

```go
prefix := []byte("coll:")
for it.Seek(prefix); it.ValidForPrefix(prefix); it.Next() {
     ...
}


```

Reply:
Oh I see. I'll adjust the code. Thank you!

Reply:
Thank you for your contribution!